curator-recipes/src/main/java/org/apache/curator/framework/recipes/queue/ChildrenCache.java

```diff
@@ -86,12 +101,14 @@ private Data(List<String> children, long version)
 
     void start() throws Exception
     {
-        sync(true);
+        client.getConnectionStateListenable().addListener(connectionStateListener);
+        sync();
     }
 
     @Override
     public void close() throws IOException
     {
+        client.getConnectionStateListenable().removeListener(connectionStateListener);
```

Comment:
Should not this be after `client.removeWatchers()`, in reverse order with `#start()`?